### ISSUE TITLE
Intermittent error when starting component due to API issue

### ISSUE THREAD

**Issue by astiskala** [sebr/bhyve-home-assistant#83]
**Describe the bug**

The Orbit B-Hyve API returned an unexpected 500 response when the component was starting, this caused the component to fail to load until Home Assistant was restarted.

**Expected behaviour**

Some sort of retry logic to handle this sort of transcient error.

**Logs**

```log
2021-09-29 22:30:31 ERROR (MainThread) [homeassistant.setup] Error during setup of component bhyve
Traceback (most recent call last):
  File "/config/custom_components/bhyve/pybhyve/client.py", line 75, in _request
    resp.raise_for_status()
  File "/usr/local/lib/python3.9/site-packages/aiohttp/client_reqrep.py", line 1000, in raise_for_status
    raise ClientResponseError(
aiohttp.client_exceptions.ClientResponseError: 500, message='Internal Server Error', url=URL('https://api.orbitbhyve.com/v1/devices?t=1632918631.2464268')

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/usr/src/homeassistant/homeassistant/setup.py", line 255, in _async_setup_component
    result = await task
  File "/config/custom_components/bhyve/__init__.py", line 120, in async_setup
    devices = [anonymize(device) for device in await bhyve.devices]
  File "/config/custom_components/bhyve/pybhyve/client.py", line 163, in devices
    await self._refresh_devices()
  File "/config/custom_components/bhyve/pybhyve/client.py", line 87, in _refresh_devices
    self._devices = await self._request(
  File "/config/custom_components/bhyve/pybhyve/client.py", line 78, in _request
    raise RequestError(f"Error requesting data from {url}: {err}")
custom_components.bhyve.pybhyve.errors.RequestError: Error requesting data from https://api.orbitbhyve.com/v1/devices: 500, message='Internal Server Error', url=URL('https://api.orbitbhyve.com/v1/devices?t=1632918631.2464268')
2021-09-29 22:30:31 ERROR (MainThread) [homeassistant.setup] Unable to prepare setup for platform bhyve.sensor: Unable to set up component.
2021-09-29 22:30:33 ERROR (MainThread) [homeassistant.setup] Unable to prepare setup for platform bhyve.switch: Unable to set up component.
```

![image](https://user-images.githubusercontent.com/12061596/135347252-2cd68f74-3dec-4ba7-aa86-e9c43e6c09c3.png)

**Comment by rtdias1970** [comment]
Hello, +1

I am getting the same error. Not so intermittent... it's happening every time.

I am running the latest version of Home Assistant (2021.10.03) on a raspberry PI 4 with 8Gb.

I can still see the bHyve entities available to Lovelace, but have no clue why the error is happening.

In any case thank you for the excellent work.

**Comment by AbhiOommen** [comment]
I am getting the same error raised above, specifically related to switching on/off the smart watering program switch, neither is the switch state updated correcting in HA.

```
Log Details (ERROR)
Logger: homeassistant.components.websocket_api.http.connection
Source: custom_components/bhyve/pybhyve/client.py:78
Integration: Home Assistant WebSocket API (documentation, issues)
First occurred: 1:19:55 PM (4 occurrences)
Last logged: 1:20:39 PM

[140415692124848] Error requesting data from https://api.orbitbhyve.com/v1/sprinkler_timer_programs/6191580e4f0c4da11bc5c304: 404, message='Not Found', url=URL('https://api.orbitbhyve.com/v1/sprinkler_timer_programs/6191580e4f0c4da11bc5c304')
Traceback (most recent call last):
  File "/config/custom_components/bhyve/pybhyve/client.py", line 75, in _request
    resp.raise_for_status()
  File "/usr/local/lib/python3.9/site-packages/aiohttp/client_reqrep.py", line 1000, in raise_for_status
    raise ClientResponseError(
aiohttp.client_exceptions.ClientResponseError: 404, message='Not Found', url=URL('https://api.orbitbhyve.com/v1/sprinkler_timer_programs/6191580e4f0c4da11bc5c304')

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/usr/src/homeassistant/homeassistant/components/websocket_api/commands.py", line 185, in handle_call_service
    await hass.services.async_call(
  File "/usr/src/homeassistant/homeassistant/core.py", line 1495, in async_call
    task.result()
  File "/usr/src/homeassistant/homeassistant/core.py", line 1530, in _execute_service
    await handler.job.target(service_call)
  File "/usr/src/homeassistant/homeassistant/helpers/entity_component.py", line 213, in handle_service
    await self.hass.helpers.service.entity_service_call(
  File "/usr/src/homeassistant/homeassistant/helpers/service.py", line 667, in entity_service_call
    future.result()  # pop exception if have
  File "/usr/src/homeassistant/homeassistant/helpers/entity.py", line 863, in async_request_call
    await coro
  File "/usr/src/homeassistant/homeassistant/helpers/service.py", line 704, in _handle_entity_call
    await result
  File "/config/custom_components/bhyve/switch.py", line 248, in async_turn_off
    await self._set_state(False)
  File "/config/custom_components/bhyve/switch.py", line 240, in _set_state
    await self._bhyve.update_program(self._program_id, self._program)
  File "/config/custom_components/bhyve/pybhyve/client.py", line 189, in update_program
    await self._request("put", path, json=json)
  File "/config/custom_components/bhyve/pybhyve/client.py", line 78, in _request
    raise RequestError(f"Error requesting data from {url}: {err}")
custom_components.bhyve.pybhyve.errors.RequestError: Error requesting data from https://api.orbitbhyve.com/v1/sprinkler_timer_programs/6191580e4f0c4da11bc5c304: 404, message='Not Found', url=URL('https://api.orbitbhyve.com/v1/sprinkler_timer_programs/6191580e4f0c4da11bc5c304')
```

**Comment by sebr** [comment]
@AbhiOommen This is a separate problem (404) - can you please raise a new issue? Thx

**Comment by sebr** [comment]
I've just released version `3.0.0-beta0` which uses the Home Assistant config flow (ie - UI based configuration). This means that any failures in the initial setup will be caught and errors will be reported in the UI with an opportunity to try again.
